compiler/rustc_privacy/src/lib.rs

```diff
@@ -1298,7 +1298,19 @@ impl SearchInterfaceForPrivateItemsVisitor<'_> {
 
     fn ty(&mut self) -> &mut Self {
         self.in_primary_interface = true;
-        self.visit(self.tcx.type_of(self.item_def_id).instantiate_identity());
+        let ty = self.tcx.type_of(self.item_def_id).instantiate_identity();
+
+        // If `in_assoc_ty`, attempt to normalize `ty`.
+        // Ideally, we would normalize in all circumstances, but doing so
+        // currently causes some unexpected type errors.
+        let maybe_normalized_ty = if self.in_assoc_ty {
+            let param_env = self.tcx.param_env(self.item_def_id);
+            self.tcx.try_normalize_erasing_regions(param_env, ty).ok()
```

Comment:
@compiler-errors you suggested on zulip that this might be replaced with a call to `normalize_erasing_regions`, but doing so caused an ICE while building libstd:
```
error: internal compiler error: compiler/rustc_middle/src/ty/normalize_erasing_regions.rs:187:90: Failed to normalize Alias(Projection, AliasTy { args: [Alias(Projection, AliasTy { args: [T/#0], def_id: DefId(0:3548 ~ core[1f97]::ops::try_trait::Try::Residual) }), V/#1], def_id: DefId(0:3559 ~ core[1f97]::ops::try_trait::Residual::TryType) }), maybe try to call `try_normalize_erasing_regions` instead

thread 'rustc' panicked at compiler/rustc_middle/src/ty/normalize_erasing_regions.rs:187:90:
Box<dyn Any>
stack backtrace:
   0: std::panicking::begin_panic::<rustc_errors::ExplicitBug>
   1: std::panic::panic_any::<rustc_errors::ExplicitBug>
   2: <rustc_errors::diagnostic::BugAbort as rustc_errors::diagnostic::EmissionGuarantee>::emit_producing_guarantee
   3: <rustc_errors::diagnostic::Diag<rustc_errors::diagnostic::BugAbort>>::emit
   4: <rustc_errors::DiagCtxt>::bug::<alloc::string::String>
   5: rustc_middle::util::bug::opt_span_bug_fmt::<rustc_span::span_encoding::Span>::{closure#0}
   6: rustc_middle::ty::context::tls::with_opt::<rustc_middle::util::bug::opt_span_bug_fmt<rustc_span::span_encoding::Span>::{closure#0}, !>::{closure#0}
   7: rustc_middle::ty::context::tls::with_context_opt::<rustc_middle::ty::context::tls::with_opt<rustc_middle::util::bug::opt_span_bug_fmt<rustc_span::span_encoding::Span>::{closure#0}, !>::{closure#0}, !>
   8: rustc_middle::ty::context::tls::with_opt::<rustc_middle::util::bug::opt_span_bug_fmt<rustc_span::span_encoding::Span>::{closure#0}, !>
   9: rustc_middle::util::bug::opt_span_bug_fmt::<rustc_span::span_encoding::Span>
  10: rustc_middle::util::bug::bug_fmt
  11: <rustc_middle::ty::normalize_erasing_regions::NormalizeAfterErasingRegionsFolder as rustc_type_ir::fold::TypeFolder<rustc_middle::ty::context::TyCtxt>>::fold_ty
  12: <rustc_middle::ty::context::TyCtxt>::normalize_erasing_regions::<rustc_middle::ty::Ty>
  13: rustc_privacy::check_private_in_public
```

Reply:
Yes, I understand why this is happening. This is because this is normalizing a not-well-formed type. I think the same thing if you write a cant-be-normalized-type in a type alias.